main/src/ca/uwaterloo/flix/language/phase/unification/Substitution.scala

```diff
@@ -57,13 +57,20 @@ case class Substitution(m: Map[Symbol.KindedTypeVarSym, Type]) {
     */
   def apply(tpe0: Type): Type = {
     // NB: The order of cases has been determined by code coverage analysis.
-    def visit(t: Type): Type =
+    // insideCaseSet means that we are inside a case set formula and should not minimize yet
+    def visit(insideCaseSet: Boolean, t: Type): Type =
```

Comment:
Can we move this visit function out of apply?

Reply:
Why are we not using the kinds to dispatch?

Reply:
The kind of the partially applied type `Apply(And, All)` does not have kind `CaseSet`. Although the `minimize` call would also not do anything since it also checks for `CaseSet` kind.

Would you prefer if I first do the normal visit, and then call a recursive minimizeCase function. This way the functions will be clearer I think

Reply:
@magnus-madsen ?